service/worker/scheduler/responsebuilder_test.go

```diff
@@ -0,0 +1,333 @@
+			assertResponseIsNil(t, response)
+		})
+	})
+
+	t.Run("when status TIMED_OUT", func(t *testing.T) {
+		rb := newResponseBuilder(&request, enumspb.WORKFLOW_EXECUTION_STATUS_TIMED_OUT, NilLogger{})
+		t.Run("when scheduled workflow result is nil returns errNoAttr", func(t *testing.T) {
+			event := historypb.HistoryEvent{Attributes: nil}
+
+			response, err := rb.Build(&event)
+
+			assertError(t, err, errNoAttrs)
+			assertResponseIsNil(t, response)
+		})
+		t.Run("when newExecutionRunId is not empty will return errFollow", func(t *testing.T) {
+			event := historypb.HistoryEvent{
+				Attributes: &historypb.HistoryEvent_WorkflowExecutionTimedOutEventAttributes{
+					WorkflowExecutionTimedOutEventAttributes: &historypb.WorkflowExecutionTimedOutEventAttributes{
+						NewExecutionRunId: "some-run-id",
+					},
+				},
+			}
+
+			response, err := rb.Build(&event)
+
+			assertError(t, err, errFollow("some-run-id"))
+			assertResponseIsNil(t, response)
+		})
+		t.Run("when newExecutionRunId is empty will return empty result", func(t *testing.T) {
+			event := historypb.HistoryEvent{
+				Attributes: &historypb.HistoryEvent_WorkflowExecutionTimedOutEventAttributes{
+					WorkflowExecutionTimedOutEventAttributes: &historypb.WorkflowExecutionTimedOutEventAttributes{},
+				},
+			}
+
+			response, err := rb.Build(&event)
+			assertError(t, err, nil)
+			assertResponseResult(t, response, nil)
+			assertResponseStatus(t, response, enumspb.WORKFLOW_EXECUTION_STATUS_TIMED_OUT)
+
+		})
+	})
+	t.Run("when status is UNSPECIFIED will return errUnknownFlow", func(t *testing.T) {
+		rb := newResponseBuilder(&request, enumspb.WORKFLOW_EXECUTION_STATUS_UNSPECIFIED, NilLogger{})
+		event := historypb.HistoryEvent{}
+
+		response, err := rb.Build(&event)
+
+		assertError(t, err, errUnkownFlow)
+		assertResponseIsNil(t, response)
+	})
+}
+
+func assertResponseResult(t *testing.T, response *schedspb.WatchWorkflowResponse, expectedResult *common.Payloads) {
+	t.Helper()
+	if response.GetResult() != expectedResult {
+		t.Errorf("expected nil result, got %v", response.GetResult())
+	}
+}
+
+func assertResponseIsNil(t *testing.T, response *schedspb.WatchWorkflowResponse) {
+	t.Helper()
+	if response != nil {
+		t.Errorf("expected response to be nil, got %v", response)
+	}
+}
+
+func assertResponsePayload(t *testing.T, response *schedspb.WatchWorkflowResponse, expectedPayload []*common.Payload) {
+	t.Helper()
+	var actualPayloads = response.GetResult().GetPayloads()
+	if !reflect.DeepEqual(actualPayloads, expectedPayload) {
+		t.Errorf("expected payload %v, got %v", expectedPayload, actualPayloads)
+	}
+}
+
+func assertResponseStatus(t *testing.T, response *schedspb.WatchWorkflowResponse, expectedStatus enumspb.WorkflowExecutionStatus) {
+	t.Helper()
+	if response.Status != expectedStatus {
+		t.Errorf("expected response expectedStatus %v, got %v", expectedStatus, response.Status)
+	}
+}
+
+func assertError(t *testing.T, err error, expectedError error) {
```

Comment:
My goal is to create easy to read and understand tests. There are more then assertError helper methods in this file. Using testify library does not make tests shorter or more expressive. Usage of if ... { t.Faillf{...} } is easy to understand to any software engineer. It is actually easier to read then require.Equal(a, b, "error message format ", a, b)

Reply:
It's about consistency: everyone working in this codebase is familiar with testify. Everyone knows to search for "Error Trace" in a long test output because that's the header that testify uses. `require.Equal(a, b)` is really quite easy to read, and that's all you need in 99% of cases. It'll print both values automatically. You can add a contextual message if you want.

Even if custom test helpers are more readable in isolation, they're not in a shared codebase with existing conventions.